### PR TITLE
Getting started

### PR DESCRIPTION
**Pull request by BrynCooke**
To test follow these instructions: https://deploy-preview-1104--apollo-router-docs.netlify.app/customizations/custom-binary

But when creating the project you'll need to specify the branch when calling cargo-scaffold.

```
cargo-scaffold scaffold https://github.com/apollographql/router.git -r apollo-router-scaffold/templates/base -t origin/bryn/getting-started --param branch=bryn/getting-started
```

### CODE REVIEW

File: scaffold/.scaffold.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+name = "apollo-router"
+author = "Apollo"
+version = "0.1.0"
+
+exclude = [
+    "./target"
+]
+
+disable_templating = [
+    "./scaffold"
```
Comment (bnjjj) [inline scaffold/.scaffold.toml:11]:
```suggestion
    "./scaffold/**/*"
```

File: apollo-router-scaffold/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+version = "0.9.2"
+authors = ["Apollo Graph, Inc. <packages@apollographql.com>"]
+edition = "2021"
+license = "Elastic-2.0"
+publish = false
+
+[dependencies]
+clap = { version = "3.1.8", features = ["derive"] }
+anyhow = "1.0.56"
+cargo-scaffold = "0.7.0"
```
Comment (bnjjj) [inline apollo-router-scaffold/Cargo.toml:12]:
```suggestion
cargo-scaffold = "0.7.1"
```

**Review by o0Ignition0o — APPROVED** [review]
This looks amazing 🎉 Just one question (I m not sure if it s a leftover or intentional; otherwise lgtm!)

File: apollo-router-scaffold/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                "snake_name".to_string(),
+                toml::Value::String(plugin_type.to_snake_case()),
+            ),
+            (
+                "pascal_name".to_string(),
+                toml::Value::String(plugin_type.to_pascal_case()),
+            ),
+            (
+                "project_name".to_string(),
+                toml::Value::String("acme".to_string()),
```
[the remark this answers is not in the document]
Comment (BrynCooke) [reply apollo-router-scaffold/src/lib.rs:95]: Yeah, it's just for the test, doesn't actually matter what the project is called, we just want to ensure that the plugin got created OK..

**Review by bnjjj — APPROVED** [review]

**Comment by o0Ignition0o** [comment]
just tried to run this and had this:

```bash
ignition@ignition-apollo starstuff % cargo b
    Updating crates.io index
    Updating git repository `https://github.com/apollographql/router.git`
    Updating git submodule `https://github.com/apollographql/supergraph-demo-fed2.git`
    Updating git submodule `https://github.com/apollographql/federation-demo.git`
error: no matching package named `apollo-router-scaffold` found
location searched: https://github.com/apollographql/router.git?tag=v0.9.2
required by package `xtask v0.1.0 (/Users/ignition/projects/oss/starstuff/xtask)`
```

**Comment by BrynCooke** [comment]
Just in case others are trying to test, you MUST run the following to scaffold and not the one in the docs:
```
cargo-scaffold scaffold https://github.com/apollographql/router.git -r apollo-router-scaffold/templates/base -t origin/bryn/getting-started --param branch=bryn/getting-started
```
This patches up the generated files to work against this branch.

**Comment by hboylan** [comment]
@BrynCooke This getting-started branch works, but it does not match the [current documentation](https://www.apollographql.com/docs/router/customizations/custom-binary#create-new-project). Is there already an issue/PR to resolve the template from the `main` branch?

**Comment by BrynCooke** [comment]
@hboylan I think that once a release happens then everything will start working. The problem is that `apollo-router-scaffold` does not exist in v0.9.2. Once v0.9.3 is released it'll be OK.

We need to fix things so that docs are pinned to the latest release rather than main, otherwise we will constantly get issues like this.

**Pull request merged**